### ISSUE TITLE
Error with VNC connections to VMWare Workstation 12

### ISSUE THREAD

**Issue by kero99** [FreeRDP/Remmina#810]
Hi community, 
after the upgrade to VMware Workstation 12.1.0 build-3272444, remmina can't connect to vnc service with this error in vmware log:

2016-03-23T10:01:32.323+01:00| mks| I125: SOCKET 102 (163) Creating VNC remote connection.
2016-03-23T10:01:32.323+01:00| mks| I125: MKSControlMgr: New VNC connection 55
2016-03-23T10:01:33.484+01:00| mks| W115: VNCENCODE 102 client wants invalid pixel format!
2016-03-23T10:01:33.484+01:00| mks| W115: VNC Error (188): The specified VNC pixel format is invalid
2016-03-23T10:01:33.484+01:00| mks| I125: SOCKET 102 (163) Deferring VNC error.
2016-03-23T10:01:33.484+01:00| mks| I125: MKSControlMgr: Remove VNC connection 55

VNC Viewer 5.2.3 work fine. But i love remmina =).

Remmina version: 1.2.0-rcgit.11

Best regards!
##

**Comment by antenore** [comment]
Can you temporary try to reduce the "Color Depth" and "Quality" settings to test if it works?

**Comment by kero99** [comment]
Hi @antenore, i test all combination of "Color Depth" and "Quality" with same result :(

2016-03-24T21:21:12.077+01:00| mks| I125: SOCKET 107 (158) Creating VNC remote connection.
2016-03-24T21:21:12.077+01:00| mks| I125: MKSControlMgr: New VNC connection 59
2016-03-24T21:21:12.569+01:00| mks| W115: VNCENCODE 107 client wants invalid pixel format!
2016-03-24T21:21:12.569+01:00| mks| W115: VNC Error (188): The specified VNC pixel format is invalid
2016-03-24T21:21:12.569+01:00| mks| I125: SOCKET 107 (158) Deferring VNC error.
2016-03-24T21:21:12.569+01:00| mks| I125: MKSControlMgr: Remove VNC connection 59

**Comment by antenore** [comment]
I don't have a VMWare at the moment to test :-(
I'll take a look the next week

**Comment by kero99** [comment]
ok!!! Thank you!!!

**Comment by antenore** [comment]
Can you please test with libvnc)?

The following should work, but let me know in the case it doesn't work

```sh
git clone https://github.com/LibVNC/libvncserver.git
cd libvncserver
./autogen.sh
cd client_examples
make
```
Than you can test the gtk client for example

Just in case, do you have a test VM I can connect to to make some tests (contact me privately my name @simbiosi.org)?

**Comment by kero99** [comment]
Hi Anton, this is the output with latest libvnc gtkclient compiled:

01/04/2016 10:46:42 PM VNC server supports protocol version 3.8 (viewer 3.8)
01/04/2016 10:46:42 PM We have 1 security types to read
01/04/2016 10:46:42 PM 0) Received security type 2
01/04/2016 10:46:42 PM Selecting security type 2 (0/1 in the list)
01/04/2016 10:46:42 PM Selected Security Scheme 2
01/04/2016 10:46:46 PM VNC authentication succeeded
01/04/2016 10:46:46 PM Desktop name "vnc-mq"
01/04/2016 10:46:46 PM Connected to VNC server, using protocol version 3.8
01/04/2016 10:46:46 PM VNC server default format:
01/04/2016 10:46:46 PM   32 bits per pixel.
01/04/2016 10:46:46 PM   Least significant byte first in each pixel.
01/04/2016 10:46:46 PM   TRUE colour: max red 255 green 255 blue 255, shift red 16 green 8 blue 01/04/2016 10:46:47 PM VNC server closed connection

Sorry i test with a machine in a company CPD and a personal laptop and i cant give you access :(, but i can test everything that you want =)

Best regards!

**Comment by piktor** [comment]
Any progress here? I have the same problem with connecting to the latest VMware Workstations 12 Player on my Ubuntu 16.04 LTS. Everything works with Vinagre, but I prefer Remmina. Thanks.

**Comment by ahttvy** [comment]
I think the problem in the vmware, but [this](https://github.com/FreeRDP/Remmina/files/611758/libvncclient_vmware.txt) patch fixes libvnc to work with VMware.

**Comment by ic3d** [comment]
Hi @vext01 @kero99 @piktor @ahttvy we're reviewing old issues, do yuo still get this one?

**Comment by ic3d** [comment]
I'm closing this issue because it probably affects an old version of Remmina. Feel free to open it again if needed.

**Comment by SilverBut** [comment]
@ic3d Hi. This issue still exists in latest Remmina, so you may want to reopen this issue. I'm current using VMware Workstation 14.0.0. And here is my version info of Remmina:
```
Failed to load plugin: /usr/lib/remmina/plugins/remmina-plugin-spice.so.
Error: libspice-client-gtk-3.0.so.5: cannot open shared object file: No such file or directory
Failed to load plugin: /usr/lib/remmina/plugins/remmina-plugin-telepathy.so.
Error: libtelepathy-glib.so.0: cannot open shared object file: No such file or directory
StatusNotifier/Appindicator support: not supported by desktop. Remmina will try to fallback to GtkStatusIcon/xembed
Remmina - 1.2.0-rcgit-26 (git rcgit-26)
NAME                TYPE            DESCRIPTION                                                     PLUGIN AND LIBRARY VERSION
EXEC                Protocol        Execute a command                                               1.0       
NX                  Protocol        NX - NX Technology                                              1.2.0-rcgit-26
RDESKTOP            Protocol        RDESKTOP - Open a RDP connection with rdesktop                  1.2.3.0   
RDP                 Protocol        RDP - Remote Desktop Protocol                                   RDP Plugin: 1.2.0-rcgit-26 (git rcgit-26), FreeRDP lib: 2.0.0-rc1 (git n/a)
RDPF                File            RDP - RDP File Handler                                          RDP Plugin: 1.2.0-rcgit-26 (git rcgit-26), FreeRDP lib: 2.0.0-rc1 (git n/a)
RDPS                Preference      RDP - Preferences                                               RDP Plugin: 1.2.0-rcgit-26 (git rcgit-26), FreeRDP lib: 2.0.0-rc1 (git n/a)
SFTP                Protocol        SFTP - Secure File Transfer                                     1.2.0-rcgit-26
SSH                 Protocol        SSH - Secure Shell                                              1.2.0-rcgit-26
VNC                 Protocol        VNC - Virtual Network Computing                                 1.2.0-rcgit-26
VNCI                Protocol        VNC - Incoming Connection                                       1.2.0-rcgit-26
XDMCP               Protocol        XDMCP - X Remote Session                                        1.2.0-rcgit-26
glibsecret          Secret          GNOME libsecret                                                 1.2.0-rcgit-26
```

**Comment by antenore** [comment]
@SilverBut can you resume the problem one more time please? I'm asking because it has passed a lot of time and I want to be sure to have the full picture, like if it works with any workaround or not at all, output from the terminal and possibly open the debug window and cut and paste also any messages.

Thanks!

Another issue, I probably don't have a VMware available for testing, anything you can do to help out?

**Comment by SilverBut** [comment]
@antenore Hi.

TL;DR: The problem has from the upstream library, `libvncserver` and fixed in the main branch but not latest release version.

If you are using `LibVNCServer-0.9.11` then you can easily reproduce this problem, just by installing any VMWare product, enable VNC and connect to it. The reason of this problem is [well described in this issue](https://github.com/LibVNC/libvncserver/issues/141) and the pull request related to [the issue is here](https://github.com/LibVNC/libvncserver/pull/170). The change is merged into the master branch, but the fixed version is not released. So the latest release version of `libvncserver` is NOT fixing this problem. And if you are using the library from your package manager (`pacman` or `apt`), the problem will happen. But if you are building `libvncserver`, you can use VNC normally.

And this bug can only be resolved by doing a binary update of `libvncserver`, so there seems not having a useful workaround, except for disable `trueColor`, which might not be a good idea.

As for the VMware....you can try to apply for a trial version of VMWare workstation. 30 days seems enough if just fixing a bug. Besides, you can also use free version such as VMWare Player. But I don't sure if the VNC function is also enabled in that version.

**Comment by antenore** [comment]
So I don't see how we can fix this except waiting for the solution or seeing if we can wrap the fixed (noway) vnc functions in remmina :-\ 
It's a pity VMWare doesn't implement the spice protocol.

Keeping this opened for those suffering from this issue.

**Comment by antenore** [comment]
BTW thanks @SilverBut

**Comment by antenore** [comment]
BTW thanks @SilverBut

**Comment by SilverBut** [comment]
Yes....just wait for new `libvncserver` and VMWare...

**Comment by antenore** [comment]
WARNING:
The Remmina project is moving to gitlab.com, in order to keep receiving notification about this issue,
you should have a gitlab account and link it to github.
The migration will happen in the next days.
Thanks for your support and understanding.

cc @ahttvy @antenore @donob4n @ic3d @kero99 @piktor @SilverBut
